ArchiSteamFarm/Json/Steam.cs

```diff
@@ -476,16 +480,171 @@ internal sealed class InventoryResponse : NumberResponse {
 			private InventoryResponse() { }
 
 			internal sealed class Description {
+				[JsonExtensionData]
+				internal readonly ImmutableDictionary<string, JToken> AdditionalProperties;
+
 				[JsonProperty(PropertyName = "appid", Required = Required.Always)]
 				internal readonly uint AppID;
 
 				internal ulong ClassID { get; private set; }
 				internal ulong InstanceID { get; private set; }
 				internal bool Marketable { get; private set; }
-				internal Asset.ERarity Rarity { get; private set; }
-				internal uint RealAppID { get; private set; }
+
+				internal Asset.ERarity Rarity {
+					get {
+						Asset.ERarity rarity = Asset.ERarity.Unknown;
+
+						foreach (Tag tag in Tags) {
+							switch (tag.Identifier) {
+								case "droprate":
+									switch (tag.Value) {
+										case "droprate_0":
+											rarity = Asset.ERarity.Common;
+
+											break;
+										case "droprate_1":
+											rarity = Asset.ERarity.Uncommon;
+
+											break;
+										case "droprate_2":
+											rarity = Asset.ERarity.Rare;
+
+											break;
+										default:
+											ASF.ArchiLogger.LogGenericError(string.Format(Strings.WarningUnknownValuePleaseReport, nameof(tag.Value), tag.Value));
+
+											break;
+									}
+
+									break;
+							}
+						}
+
+						return rarity;
+					}
+				}
+
+				internal uint RealAppID {
+					get {
+						uint realAppID = 0;
+
+						foreach (Tag tag in Tags) {
+							switch (tag.Identifier) {
+								case "Game":
+									if ((tag.Value.Length <= 4) || !tag.Value.StartsWith("app_", StringComparison.Ordinal)) {
+										ASF.ArchiLogger.LogGenericError(string.Format(Strings.WarningUnknownValuePleaseReport, nameof(tag.Value), tag.Value));
+
+										break;
+									}
+
+									string appIDText = tag.Value.Substring(4);
+
+									if (!uint.TryParse(appIDText, out uint appID) || (appID == 0)) {
+										ASF.ArchiLogger.LogNullError(nameof(appID));
+
+										break;
+									}
+
+									realAppID = appID;
+
+									break;
+							}
+						}
+
+						return realAppID;
+					}
+				}
+
+				[JsonProperty(PropertyName = "tags", Required = Required.DisallowNull)]
+				internal ImmutableHashSet<Tag> Tags;
+
 				internal bool Tradable { get; private set; }
-				internal Asset.EType Type { get; private set; }
+
+				internal Asset.EType Type {
+					get {
+						Asset.EType type = Asset.EType.Unknown;
+
+						foreach (Tag tag in Tags) {
+							switch (tag.Identifier) {
+								case "cardborder":
+									switch (tag.Value) {
+										case "cardborder_0":
+											type = Asset.EType.TradingCard;
```

Comment:
This is nice, it looks now how I've imagined it initially.

Now you can slightly improve performance of those getters, you can right away `return` the appropriate value instead of saving it, and falling back to `return Unknown` or other default in case no tag matches the expected name.